src/main/java/com/google/api/generator/engine/ast/StringObjectValue.java

```diff
@@ -37,6 +40,38 @@ public TypeNode type() {
   public abstract static class Builder {
     public abstract Builder setValue(String value);
 
-    public abstract StringObjectValue build();
+    public abstract StringObjectValue autoBuild();
+
+    public StringObjectValue build() {
+      // `\"` is added to the escaped string value for interpreting it correctly in file.
+      String value =
+          String.format("\"%s\"", StringValueEscaper.getInstance().escape(autoBuild().value()));
```

Comment:
Instead of calling `autoBuild` twice, what about just using `value()`? IIRC we discussed this in a prior review.

Reply:
right, we had this discussion, here we cannot use `value()` because it's not static, Builder class has static context.
And pasted some original explanation for the reason why not removing the second autoBuild():

I think calling autoBuild twice is necessary, since if the object is built we cannot re-set the value (autovalue pattern) unless we re-create one (second autoBuild).
if we set the value by below to avoid the second autoBuild, there will be OutOfMemoryError, coz it will be a loop build() -> setvalue() -> build()
```
Builder builder= setValue(value);
return builder.build();
```

Reply:
One way to avoid calling `autoBuild()` twice is to add a "private" `value()` method to the Builder class. Please see the other examples in this codebase.

Reply:
Right! Thank you for the advice, also found this [doc](https://github.com/google/auto/blob/master/value/userguide/builders-howto.md#normalize) that illustrate how/why it works. Learned from this PR!